src/net/server/connection.rs

```diff
@@ -682,112 +690,24 @@ where
-
-                                        ServiceFeedback::EndTransaction => {
-                                            in_transaction = false;
-                                        }
-                                    }
-                                }
-
-                                if let Some(mut response) = response {
-                                    loop {
-                                        match result_q_tx.try_send(response) {
-                                            Ok(()) => {
-                                                let pending_writes =
-                                                    result_q_tx
-                                                        .max_capacity()
-                                                        - result_q_tx
-                                                            .capacity();
-                                                trace!("Queued message for sending: # pending writes={pending_writes}");
-                                                metrics
-                                                    .set_num_pending_writes(
-                                                        pending_writes,
-                                                    );
-                                                break;
-                                            }
-
-                                            Err(TrySendError::Closed(_)) => {
-                                                error!("Unable to queue message for sending: server is shutting down.");
-                                                break;
-                                            }
-
-                                            Err(TrySendError::Full(
-                                                unused_response,
-                                            )) => {
-                                                if in_transaction {
-                                                    // Wait until there is space in the message queue.
-                                                    tokio::task::yield_now()
-                                                        .await;
-                                                    response =
-                                                        unused_response;
-                                                } else {
-                                                    error!("Unable to queue message for sending: queue is full.");
-                                                    break;
-                                                }
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                            trace!("Finished processing service call results for request id {request_id}");
+                            dispatcher.dispatch(request, service, ()).await
```

Comment:
I get the impression that if Message::from_octets fails the client doesn't get a reply